### PR TITLE
Add types to function

### PR DESCRIPTION
**Pull request by desoindx**

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -65,3 +68,8 @@ overrides:
     rules:
       max-len: "off"
       func-names: "off"
+  - files:
+      - "test/*.ts"
+      - "*.js"
+    rules:
+      "@typescript-eslint/explicit-function-return-type": "off"
```
Comment (sandrica89) [inline .eslintrc.yml:75]: Why not put this rule after l.65 for the files: 
```
      - "tests/*.ts"
      - "test/*.ts"
      - "*.spec.ts"
```

Also, not sure `tests/*.ts` and `*.js` are still necessary

File: services/demarchesSimplifiees/importDossier.ts [hunk 2]
```diff
@@ -21,10 +22,17 @@ const getUuidDossierNumber = (number: number): string => (
...
-    state, archived, demandeur, usager, number, groupeInstructeur, champs,
+    state,
+    archived,
+    demandeur,
+    usager,
+    number,
+    groupeInstructeur,
+    champs,
   } = dossier;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (sandrica89) [inline services/demarchesSimplifiees/importDossier.ts:35]: Should not be needed anymore

File: test/loginController.spec.ts [hunk 3]
```diff
@@ -321,20 +322,20 @@ describe('loginController', async () => {
...
         .get('/api/connecteduser')
-        .set('Cookie', `token=${cookie.getJwtTokenForUser(uuidv4())}`)
+        .set('Cookie', `token=${cookie.getJwtTokenForUser(uuidv4(), 'randomXSRFToken')}`)
         .then(async (res) => res.body.should.be.empty));
 
     it('should return empty info if user is not connected', async () => chai
         .request(app)
         .get('/api/connecteduser')
-        .then(async (res) => res.body.should.be.empty));
+        .then(async (res) => { console.log(res.body); res.body.should.be.empty; }));
```
Comment (sandrica89) [inline test/loginController.spec.ts:331]: Is this a log for dev purpose?

File: controllers/loginController.ts [hunk 4]
```diff
@@ -179,4 +179,6 @@ export default {
   login: asyncHelper(login),
   sendMail: asyncHelper(sendMail),
   deleteToken,
+  generateLoginUrl,
+  generateToken,
```
Comment (sandrica89) [inline controllers/loginController.ts:183]: It seems that these methods were exported for test purpose, they are never called.
We may refactor this => put them in a service. If not done here (already too big PR), create a task to do it later.

File: db/appointments.ts [hunk 5]
```diff
@@ -1,125 +1,22 @@
...
-// this function returns empty.
-const getAll = async (psychologistId: string): Promise<any[]> => {
+const getAll = async (psychologistId: string): Promise<AppointmentWithPatient[]> => {
   try {
     const appointmentArray = await db.from(patientsTable)
-      .innerJoin(`${appointmentsTable}`, `${patientsTable}.id`, `${appointmentsTable}.patientId`)
-      .where(`${appointmentsTable}.psychologistId`, psychologistId)
-      .whereNot(`${appointmentsTable}.deleted`, true)
-      .orderBy('appointmentDate', 'desc');
+    .innerJoin(`${appointmentsTable}`, `${patientsTable}.id`, `${appointmentsTable}.patientId`)
```
Comment (sandrica89) [inline db/appointments.ts:9]:
```suggestion
    .innerJoin(appointmentsTable, `${patientsTable}.id`, `${appointmentsTable}.patientId`)
```

File: db/appointments.ts [hunk 6]
```diff
@@ -149,6 +46,7 @@ const deleteOne = async (appointmentId: string, psychologistId: string): Promise
 
     console.log(`Appointment id ${appointmentId} deleted by psy id ${psychologistId}`);
 
+    console.log('deleted', deletedAppointments);
```
Comment (sandrica89) [inline db/appointments.ts:49]: Debug log?

File: scripts/insertEmailToUniversities.ts [hunk 7]
```diff
@@ -59,7 +60,7 @@ const parseFile = () => {
         }),
         {},
       ));
-    insertEmailToUniversities(data);
+    insertEmailToUniversities(data as University[]);
```
Comment (sandrica89) [inline scripts/insertEmailToUniversities.ts:63]: Can throw exception if cast doesn't work?

File: test/autoVerifyPsychologist.spec.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    sinon.assert.callCount(verifyPsychologistStub, 2);
+
+    sinon.assert.calledWith(verifyPsychologistStub.getCall(0),
+      sinon.match(),
+      sinon.match());
+
+    sinon.assert.calledWith(verifyPsychologistStub.getCall(1),
+      sinon.match(),
+      sinon.match());
+  }).timeout(30000);
```
Comment (sandrica89) [inline test/autoVerifyPsychologist.spec.ts:30]: The test was not set after refactoring of autoVerify. Do you want me to do it now OR will we fix it when adeli API will be back?

File: test/cookie.spec.ts [hunk 9]
```diff
@@ -1,38 +1,35 @@
...
   });
 
   describe('getJwtTokenForUser', () => {
     it('should return a json web token with only id', () => {
       const psychologist = uuidv4();
-      const token = cookie.getJwtTokenForUser(psychologist);
+      const token = cookie.getJwtTokenForUser(psychologist, 'randomXSRF');
 
-      token.length.should.be.equal(200);
+      token.length.should.be.equal(233);
```
Comment (sandrica89) [inline test/cookie.spec.ts:21]: :thinking: isn't that weird that it changed?

File: types/Appointment.ts [hunk 10]
```diff
@@ -1,7 +1,11 @@
...
+
 export type Appointment = {
     id: string,
     psychologistId: string,
     appointmentDate: string,
     patientId: string,
     deleted: boolean
 }
+
+export type AppointmentWithPatient = Appointment & Patient;
```
Comment (sandrica89) [inline types/Appointment.ts:11]: :thinking: shouldn't this done on the specific place where it's used? It's not a 'real' type.

File: utils/sendGraphQLRequest.ts [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+  if (apiResponse.response) {
+    if (apiResponse.response.errors.length > 0) {
+      apiResponse.response.errors.forEach((err) => {
+        console.error('Error details', err);
+      });
+    }
+  }
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```
Comment (sandrica89) [inline utils/sendGraphQLRequest.ts:26]: Shouldn't we use DSReponse here and remove this comment (and the others below)?

**Review by sandrica89 — CHANGES_REQUESTED** [review]
The tests were moved to `test/*` because there were not working in subdirectories. Was this linked to rewire? Can we put them back in subdirectories?

File: scripts/insertEmailToUniversities.ts (hunk 7 above)
Comment (desoindx) [reply scripts/insertEmailToUniversities.ts:63]: no, casting is only used on compilation, no runtime exception can be throwns

File: test/autoVerifyPsychologist.spec.ts (hunk 8 above)
Comment (desoindx) [reply test/autoVerifyPsychologist.spec.ts:30]: we'll fix it later

File: test/cookie.spec.ts (hunk 9 above)
Comment (desoindx) [reply test/cookie.spec.ts:21]: no, now the getJwtTokenForUser force a xsrf token on second parameters, so it's added to the token

File: utils/sendGraphQLRequest.ts (hunk 11 above)
Comment (desoindx) [reply utils/sendGraphQLRequest.ts:26]: no because we have other queries (like GroupeInstructeur)

**Comment by desoindx** [comment]
> The tests were moved to `test/*` because there were not working in subdirectories. Was this linked to rewire? Can we put them back in subdirectories?

Not related to rewire but to ts-mocha, no idea why :(
you can try if you like :)

**Pull request merged**